onmt/IO.py

```diff
@@ -275,9 +304,14 @@ class ONMTDataset(torchtext.data.Dataset):
     @staticmethod
     def sort_key(ex):
         "Sort in reverse size order"
-        return -len(ex.src)
+        if hasattr(ex, 'src'):
```

Comment:
I don't like the use of `hasattr`, that's a magic method. Please add an actual member to check.